pyuvdata/uvdata.py

```diff
@@ -616,14 +614,15 @@ def phase(self, ra, dec, epoch):
         # explicitly set epoch to J2000
         self.phase_center_epoch = 2000.0
 
+        c_s = const.c.to('m/s').value
         for ind, jd in enumerate(self.time_array):
             # calculate ra/dec of phase center in current epoch
             obs.date, obs.epoch = self.juldate2ephem(jd), self.juldate2ephem(jd)
             precess_pos.compute(obs)
             ra, dec = precess_pos.a_ra, precess_pos.a_dec
 
             # generate rotation matrices
-            m0 = a.coord.top2eq_m(self.lst_array[ind] - obs.sidereal_time(), latitude)
+            m0 = a.coord.top2eq_m(self.lst_array[ind] - obs.sidereal_time(), latitude)    ##TODO This is a bottleneck because top2eq_m involves a matrix inversion. 
```

Comment:
I don't have time to do it now, but one way to at least reduce the bottleneck, would be to only calculate m0 for unique times, rather than for all blts.